src/main/java/com/github/manolo8/darkbot/gui/MainGui.java

```diff
@@ -33,8 +37,24 @@ public MainGui(Main main) throws HeadlessException {
         ToolTipManager.sharedInstance().setInitialDelay(350);
 
         setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
-        setSize(640, 480);
-        setLocationRelativeTo(null);
+
+        Config.BotSettings botSettings = main.config.BOT_SETTINGS;
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            if (botSettings.SAVE_MAIN_GUI_POS_AND_SIZE)
+                main.configManager.saveConfig();
+        }));
+
+        Config.BotSettings.Window thisWindow = botSettings.MAIN_GUI_WINDOW;
```

Comment:
using the word *this* in a variable name is a bit weird to see. call it winLoc or location or position or something along the lines